### ISSUE TITLE
Unable to generate firmware for basic installation

### ISSUE THREAD

**Issue by tajacks** [fhunleth/nerves_system_rpi0_2#46]
Hello - 

I am trying to do the "bare minimum" to get this running on a Raspi Zero 2 W I have. Unfortunately I have not had success, and, it's unclear to me what additional steps (if any) I need to take. From the top:

`~/code_temp ❱ mix nerves.new nerves_tester`

This results in the following `mix.exs`, less comments for brevity:

```
defmodule NervesTester.MixProject do
  use Mix.Project

  @app :nerves_tester
  @version "0.1.0"
  @all_targets [
    :rpi,
    :rpi0,
    :rpi2,
    :rpi3,
    :rpi3a,
    :rpi4,
    :rpi5,
    :bbb,
    :osd32mp1,
    :x86_64,
    :grisp2,
    :mangopi_mq_pro
  ]

  def project do
    [
      app: @app,
      version: @version,
      elixir: "~> 1.17",
      archives: [nerves_bootstrap: "~> 1.13"],
      start_permanent: Mix.env() == :prod,
      deps: deps(),
      releases: [{@app, release()}],
      preferred_cli_target: [run: :host, test: :host]
    ]
  end

  def application do
    [
      extra_applications: [:logger, :runtime_tools],
      mod: {NervesTester.Application, []}
    ]
  end

  defp deps do
    [
      # Dependencies for all targets
      {:nerves, "~> 1.10", runtime: false},
      {:shoehorn, "~> 0.9.1"},
      {:ring_logger, "~> 0.11.0"},
      {:toolshed, "~> 0.4.0"},
      {:nerves_runtime, "~> 0.13.0"},
      {:nerves_pack, "~> 0.7.1", targets: @all_targets},
      {:nerves_system_rpi, "~> 1.24", runtime: false, targets: :rpi},
      {:nerves_system_rpi0, "~> 1.24", runtime: false, targets: :rpi0},
      {:nerves_system_rpi2, "~> 1.24", runtime: false, targets: :rpi2},
      {:nerves_system_rpi3, "~> 1.24", runtime: false, targets: :rpi3},
      {:nerves_system_rpi3a, "~> 1.24", runtime: false, targets: :rpi3a},
      {:nerves_system_rpi4, "~> 1.24", runtime: false, targets: :rpi4},
      {:nerves_system_rpi5, "~> 0.2", runtime: false, targets: :rpi5},
      {:nerves_system_bbb, "~> 2.19", runtime: false, targets: :bbb},
      {:nerves_system_osd32mp1, "~> 0.15", runtime: false, targets: :osd32mp1},
      {:nerves_system_x86_64, "~> 1.24", runtime: false, targets: :x86_64},
      {:nerves_system_grisp2, "~> 0.8", runtime: false, targets: :grisp2},
      {:nerves_system_mangopi_mq_pro, "~> 0.6", runtime: false, targets: :mangopi_mq_pro}
    ]
  end

  def release do
    [
      overwrite: true,
      cookie: "#{@app}_cookie",
      include_erts: &Nerves.Release.erts/0,
      steps: [&Nerves.Release.init/1, :assemble],
      strip_beams: Mix.env() == :prod or [keep: ["Docs"]]
    ]
  end
end
```

Without making any modifications, and setting my target via `export MIX_TARGET=rpi0_2`,  I can get dependencies, but attempting to run `mix firmware` results in:

```
==> nerves_tester
** (Mix) Environment variable $NERVES_SYSTEM is not set.

This variable is usually set for you by Nerves when you specify the
$MIX_TARGET. It is unusual to need to specify it yourself.

Some things to check:

1. In your `mix.exs`, is the value that you have in $MIX_TARGET in the
  `@all_targets` list? If you're not using `@all_targets`, then the
  $MIX_TARGET should appear in the `:targets` option for `:nerves_runtime`
  and other packages that run on the target.

2. Do you have a dependency on a Nerves system for the target? For example,
  `{:nerves_system_rpi0, "~> 1.8", runtime: false, targets: :rpi0}`

3. Is there a typo? For example, is $MIX_TARGET set to `rpi1` when it should
  be `rpi`.

4. Is there a typo in the package name of the system? For example, if you
  have a custom system, `:nerves_system_my_board`, does the spelling of the
  system in the dependency in your `mix.exs` match the spelling in your
  system project's `mix.exs`?

For build examples in the Nerves documentation, please see
https://hexdocs.pm/nerves/getting-started.html#create-the-firmware-bundle
```

Ok, so perhaps I need to add it to the targets list. Doing so and running `mix firmware` again results in:

```
==> vintage_net_wifi
Package libnl-genl-3.0 was not found in the pkg-config search path.
Perhaps you should add the directory containing `libnl-genl-3.0.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libnl-genl-3.0' found
 CC force_ap_scan.o
src/force_ap_scan.c:20:10: fatal error: netlink/genl/genl.h: No such file or directory
   20 | #include <netlink/genl/genl.h>
      |          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:84: /home/tajacks/code_temp/nerves_tester/_build/rpi0_2_dev/lib/vintage_net_wifi/obj/force_ap_scan.o] Error 1
could not compile dependency :vintage_net_wifi, "mix compile" failed. Errors may have been logged above. You can recompile this dependency with "mix deps.compile vintage_net_wifi --force", update it with "mix deps.update vintage_net_wifi" or clean it with "mix deps.clean vintage_net_wifi"
==> nerves_tester
** (Mix) Could not compile with "make" (exit status: 2).
```

Upon seeing this, I installed libnl-genl-3.0 on my host. Back to first failure case. Do I need to add the dependency?

In short, what are the steps needed to get up and running with a basic nerves project, including modification from the project generator, so I can flash this to run on my Pi Zero 2W?

**Comment by fhunleth** [comment]
Due to a some long history, the Raspberry Pi Zero 2W's system for Nerves is `nerves_system_rpi3a`. Ie., run `export MIX_TARGET=rpi3a`, then `mix deps.get`, then `mix firmware`.

This system is a 64-bit version of Nerves for the Pi Zero 2W. Technically it's still experimental, but it should work too. If you're beginning with Nerves, I don't think there's a compelling reason to use it over `:rpi3a`. Erlang works faster with 64-bit, but I wouldn't expect it to be noticeable until you get a lot running on it. Phoenix, for example, should still be pretty quick at 32-bits.

This system is not included in the `mix.exs` that `nerves.new` generates. You'll have to add the following to your deps:

```
    {:nerves_system_rpi0_2, "~> 1.28", runtime: false, targets: :rpi0_2},
```

And then at the top, add `:rpi0_2` to the all targets list:

```
   @all_targets  [
        ...
        :rpi0_2
      ]
 ```

**Comment by tajacks** [comment]
Well I was pretty sure I tried that but evidently not, must have been a PEBKAC error, because I was able to get it to generate the firmware (and run) in the end. Thanks!

`nerves_tester_rpi0 0.1.0 (d6ba8597-7bd0-57a4-06a7-71cfdafcd1c0) aarch64 rpi0_2`

I assume that the intent is at one point to make this nerves system the default for rpi02/W ?

**Comment by fhunleth** [comment]
Yes, that's the plan. The camera and display aren't well tested, but it seems good so far.
